common/plugins/org.jboss.tools.common.jdt.debug/src/org/jboss/tools/common/jdt/debug/tools/ToolsCore.java

```diff
@@ -95,7 +96,17 @@ public Object getMonitoredVM() {
 	}
 	
 	public static boolean processIsRunning(String hostname, int vmPid) throws ToolsCoreException {
-		return Tools.getInstance().getMonitoredVm(hostname, vmPid) != null;
+		// If these throw exceptions, we want them to be rethrown
+		Object host = Tools.getInstance().invokeGetMonitoredHost(hostname);
+		Object vmId = Tools.getInstance().invokeVmIdentifier(String.format(IToolsConstants.VM_IDENTIFIRER, vmPid));
+		
+		// We want to catch and ignore these failures and just return that the process is not running
+		try {
+			Object monitoredVm = Tools.getInstance().invokeGetMonitoredVm(host, vmId);
+			return monitoredVm != null;
+		} catch(ToolsCoreException tce) {
```

Comment:
The problem is that these exceptions, if logged, could potentially be logged every x seconds. In almost all of these cases, it is best to simply ignore the process that suffers from *ANY* of these exceptions. 

Only one method calls processIsRunning, and that is RemoteDebugActivator.getVmModelsUsingTools().   This method is meant to create a model of things we *can* monitor. It is no benefit to us to list or pay attention to things we cannot monitor. 

If you're suggesting we should include *all* processes that are running into our model, then that would require substantial changes throughout the rest of the code to then further check whether they can be monitored. 

I can add a comment that indicates processIsRunning() would be more correctly named processIsMonitorable() to be more clear.